plugins/org.python.pydev/tests_completions/org/python/pydev/ast/codecompletion/PythonCompletionWithoutBuiltinsGrammar3Test.java

```diff
@@ -287,4 +287,178 @@ public void testTypingCast() throws Exception {
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "def method1():\n"
+                + "    param: foo.Union[A, B]\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()" });
+    }
+
+    public void testUnion8() throws Exception {
+        String s;
+        s = "" +
+                "from typing import Union\n"
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "def method1(param: A | B):\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()" });
+    }
+
+    public void testUnion9() throws Exception {
+        String s;
+        s = "" +
+                "from typing import Union\n"
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "class C(object):\n"
+                + "    def method_c(self):\n"
+                + "        pass\n"
+                + "def method1(param: A | B | C):\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()", "method_c()" });
+    }
+
+    public void testUnion10() throws Exception {
+        String s;
+        s = "" +
+                "from typing import Union\n"
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "def method1(param:A|B):\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()" });
+    }
+
+    public void testUnion11() throws Exception {
+        String s;
+        s = "" +
+                "from typing import Union\n"
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "class C(object):\n"
+                + "    def method_c(self):\n"
+                + "        pass\n"
+                + "def method1(param:A|B|C):\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()", "method_c()" });
```

Comment:
I'd like to add a test for a case such as:

```
def method1(param: List[A|B]):
    for x in param:
        x.  # at this point it should present completions for A and B.
```

Reply:
p.s.: Same for `param: List[Union[A, B]]`.

Reply:
Did it ;) Check out the latest commits